### PR TITLE
add receive modal

### LINKED ISSUE #90: shows the corresponding page when the general app receive page is clicked
Screenshot 2023-09-25 at 2 44 55 AM

### PR DESCRIPTION
**Pull request by dreacot**
closes #90 

Screenshot from 2023-09-28 15-35-56

### CODE REVIEW

File: ui/page/components/receive_modal.go [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+		rm.isNewAddr = !rm.isNewAddr
+		if rm.isInfo {
+			rm.isInfo = false
+		}
+	}
+
+	if rm.newAddr.Clicked() {
+		newAddr, err := rm.generateNewAddress()
+		if err != nil {
+			log.Debug("Error generating new address" + err.Error())
```
Comment (ukane-philemon) [inline ui/page/components/receive_modal.go:129]: nit: needs space before the closing quotation mark.

Comment (ukane-philemon) [inline ui/page/components/receive_modal.go:129]: nit: needs space before the closing quotation mark.

File: ui/page/components/receive_modal.go [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+	var buffer bytes.Buffer
+	err = qrImage.SaveTo(&buffer)
+	if err != nil {
+		log.Error(err.Error())
+		return
+	}
+
+	decodeImg, _, err := image.Decode(bytes.NewReader(buffer.Bytes()))
+	if err != nil {
+		log.Error(err.Error())
```
Comment (ukane-philemon) [inline ui/page/components/receive_modal.go:183]: Same as my comment above.

File: ui/page/components/receive_modal.go [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+
+generateAddress:
+	newAddr, err := rm.sourceWalletSelector.selectedWallet.NextAddress(rm.sourceAccountSelector.selectedAccount.Number)
+	if err != nil {
+		return "", err
+	}
+
+	if newAddr == rm.currentAddress {
+		goto generateAddress
+	}
```
Comment (ukane-philemon) [inline ui/page/components/receive_modal.go:200]: Is this possible?

**Review by ukane-philemon — CHANGES_REQUESTED** [review]
Testing well except for a few touchups needed:

The copy toast alert does not sit well with the current position, I think. Maybe below the qr code? There's some estate there.

Screenshot 2023-09-28 at 11 00 50 PM

The Generate Address btn becomes transparent on hover, is that intended?

Also, I think clicking outside of the Generate Address btn should close the more display, no?
Screenshot 2023-09-28 at 11 00 26 PM

File: ui/page/components/receive_modal.go (hunk 3 above)
Comment (dreacot) [reply ui/page/components/receive_modal.go:200]: yes!, magic of go

Comment (ukane-philemon) [reply ui/page/components/receive_modal.go:200]: Yh. I meant to confirm if this case `newAddr == rm.currentAddress` is ever possible, if not we can skip it.

Comment (dreacot) [reply ui/page/components/receive_modal.go:200]: oh, my bad, yes there's a bug we discovered a while back, so this was a hack to bypass it, https://github.com/planetdecred/dcrlibwallet/pull/241/files the bug was fixed here but the code of the receive page was never updated, i ported over code from the receive page, i'd update it now, good catch

**Comment by dreacot** [comment]
> The copy toast alert does not sit well with the current position, I think. Maybe below the qr code? There's some estate there.

This is displayed using `rm.Load.Toast.Notify(values.String(values.StrCopied))` there's no way to change the position except the Notify method is modified to receive a position variable, which i think would be overkill for this PR

**Comment by ukane-philemon** [comment]
> `ify(values.String(values.StrCopied))` there's no way to change the position except the Notify method is modified to receive a position variable, which i think would be overkill for this PR

Okay.

**Review by ukane-philemon — APPROVED** [review]
Testing well.

**Pull request merged**